dashboard/app/models/user.rb

```diff
@@ -157,6 +157,12 @@ class User < ActiveRecord::Base
   accepts_nested_attributes_for :school_info, reject_if: :preprocess_school_info
   validates_presence_of :school_info, unless: :school_info_optional?
 
+  validate :must_be_a_teacher_if_has_sections, on: :update, if: -> {user_type_changed? && student?}
```

Comment:
Not sure.. My main concern is that I don't understand all of the impacts of requiring `section.valid?`, given that lots of pre-existing sections fail existing validations for various reasons. If you feel we should go ahead with the change (as-is), I'm happy to think through its implications and review (as-is).